sklearnex/spmd/decomposition/tests/test_incremental_pca_spmd.py

```diff
@@ -0,0 +1,266 @@
+        local_dpt_X = _convert_to_dataframe(
+            split_local_X[i], sycl_queue=queue, target_df=dataframe
+        )
+        dpt_X = _convert_to_dataframe(X_split[i], sycl_queue=queue, target_df=dataframe)
+        incpca.partial_fit(dpt_X)
+        incpca_spmd.partial_fit(local_dpt_X)
+
+    assert_allclose(incpca.n_components_, incpca_spmd.n_components_)
+    assert_allclose(incpca.components_, incpca_spmd.components_)
+    assert_allclose(incpca.singular_values_, incpca_spmd.singular_values_)
+    assert_allclose(incpca.mean_, incpca_spmd.mean_)
+    assert_allclose(incpca.var_, incpca_spmd.var_)
+    assert_allclose(incpca.explained_variance_, incpca_spmd.explained_variance_)
+    assert_allclose(
+        incpca.explained_variance_ratio_, incpca_spmd.explained_variance_ratio_
+    )
+
+
+@pytest.mark.skipif(
+    not _mpi_libs_and_gpu_available,
+    reason="GPU device and MPI libs required for test",
+)
+@pytest.mark.parametrize(
+    "dataframe,queue",
+    get_dataframes_and_queues(dataframe_filter_="dpnp,dpctl", device_filter_="gpu"),
+)
+@pytest.mark.parametrize("whiten", [True, False])
+@pytest.mark.parametrize("n_components", [None, 2, 5])
+@pytest.mark.parametrize("num_samples", [100, 200])
+@pytest.mark.parametrize("num_features", [10, 20])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+@pytest.mark.mpi
+def test_incremental_pca_fit_spmd_random(
+    dataframe, queue, whiten, n_components, num_samples, num_features, dtype
+):
+    # Import spmd and non-SPMD algo
+    from sklearnex.preview.decomposition import IncrementalPCA
+    from sklearnex.spmd.decomposition import IncrementalPCA as IncrementalPCA_SPMD
+
+    tol = 7e-5 if dtype == np.float32 else 1e-7
+
+    # Create data and process into dpt
+    X = _generate_statistic_data(num_samples, num_features, dtype)
+    dpt_X = _convert_to_dataframe(X, sycl_queue=queue, target_df=dataframe)
+    X_test = _generate_statistic_data(num_samples // 5, num_features, dtype)
+    dpt_X_test = _convert_to_dataframe(X_test, sycl_queue=queue, target_df=dataframe)
+    local_X = _get_local_tensor(X)
+    local_dpt_X = _convert_to_dataframe(local_X, sycl_queue=queue, target_df=dataframe)
+
+    incpca_spmd = IncrementalPCA_SPMD(n_components=n_components, whiten=whiten)
+    incpca = IncrementalPCA(n_components=n_components, whiten=whiten)
+
+    incpca_spmd.fit(local_dpt_X)
+    incpca.fit(dpt_X)
+
+    assert_allclose(incpca.n_components_, incpca_spmd.n_components_, atol=tol)
+    assert_allclose(incpca.components_, incpca_spmd.components_, atol=tol)
+    assert_allclose(incpca.singular_values_, incpca_spmd.singular_values_, atol=tol)
+    assert_allclose(incpca.mean_, incpca_spmd.mean_, atol=tol)
+    assert_allclose(incpca.var_, incpca_spmd.var_, atol=tol)
+    assert_allclose(incpca.explained_variance_, incpca_spmd.explained_variance_, atol=tol)
+    assert_allclose(
+        incpca.explained_variance_ratio_, incpca_spmd.explained_variance_ratio_, atol=tol
+    )
```

Comment:
How come this one needs a special tolerance setting and the others above don't?
```suggestion
    for attr in incpca:
        if attr.endswith("_") and not attr.startswith("_"):
            assert_allclose(getattr(incpca, attr), getattr(incpa_spmd, attr), err_msg=attr, atol=tol)
```
```

Reply:
because this one uses bigger data and it leads to accumulation of bigger computational error

Reply:
sounds like a good comment to add above the definition of tol

Reply:
i thought it is clear, we use different tolerance for different dtype in almost all tests. why should it be commented only here?

Reply:
Its notable because the other tests didn't require setting it.  I think if its not obvious or notable to me but can't be explained, its best to leave information for the people after you. Documentation is your friend.

Reply:
i don't really know what to put in the comment. "tolerance for float32 is more here because float32 computations are less accurate"? isn't it obvious why it didn't required? let's save comments for places which are really hard to understand. we have a lot of tests with tolerance setting and a lot of tests without it. i don't know if we have any standards about that at all.

Reply:
ok, done